geoh5py/shared/concatenation/concatenator.py

```diff
@@ -615,3 +618,131 @@ def update_array_attribute(
             self.data[alias] = values
 
         self.save_attribute(field)
+
+    @staticmethod
+    def get_depth_association(
+        property_group: ConcatenatedPropertyGroup,
+    ) -> tuple[str] | tuple[str, str] | tuple[None]:
+        """
+        Based on a PropertyGroup, it gets the name of the depth (or from to)
+            associated with the data.
+
+        :param property_group: The property group to extract.
+
+        return: the name of the values to used for association.
+        """
+        if getattr(property_group, "property_group_type", None) == "Interval table":
+            return property_group.from_.name, property_group.to_.name
+        if getattr(property_group, "property_group_type", None) == "Depth table":
+            return (property_group.depth_.name,)
+
+        return (None,)
+
+    @staticmethod
+    def get_properties_names(property_group: ConcatenatedPropertyGroup) -> tuple:
+        """
+        Get the names of the properties in the property group.
+
+        :param property_group: The property group to extract.
+        """
+        if property_group.properties is None:
+            return (None,)
+        return tuple(
+            property_group.parent.get_data(property_)[0].name
+            for property_ in property_group.properties
+        )
+
+    @property
+    def unique_property_group_names(self) -> dict:
```

Comment:
I think it would be nicer to ask for all Tables, and let Tables figure out what is under them.